Nodejs/Product/TestAdapter/TestFrameworks/mocha/mocha.js

```diff
@@ -111,7 +111,7 @@ var run_tests = function (testCases, callback) {
     mocha.addFile(testCases[0].testFile);
 
     var runner = mocha.run(function (code) {
-        process.exit(code);
+        process.exitCode = code ? code : 0;
```

Comment:
So this no longer exits the process when called, just sets the exit code for when it exits later by some other mechanism. Is that right?

Reply:
That's correct. With this NodeJs will close itself after all operations have finished instead of killing and truncating the stdout.